x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/connector_add_flyout.tsx

```diff
@@ -199,10 +203,21 @@ const ConnectorAddFlyout: React.FunctionComponent<ConnectorAddFlyoutProps> = ({
         );
         return;
       }
+
       setIsSaving(true);
+      // Do not allow to save the connector if there is an error
+      try {
+        await callbacks?.beforeActionConnectorSave?.();
```

Comment:
Call the connector's callback before saving the connector.

Reply:
Oh that's cool, does the `?.()` only call it if it is defined?

Reply:
Yes!  I think I don't use it enough.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining